app/bundles/EmailBundle/Model/EmailModel.php

```diff
@@ -938,7 +938,8 @@ public function getPendingLeads(
             $maxContactId,
             $countWithMaxMin,
             $maxThreads,
-            $threadId
+            $threadId,
+            $email->isSegmentEmail() && !$email->getContinueSending() ? $email->getPublishUp() : null,
```

Comment:
@kuzmany I am writing the missing test but I don't understand this line:
 why do we give $email->getPublishUp() to $sendStopDate parameter ? Perhaps it should be $email->getPublishDown() ?

Reply:
We want to send emails only to contacts that we added before we start sending emails. This is the correct approach.